### DISCUSSION TITLE
A way to shift animation data

### DISCUSSION THREAD

**Discussion by aganm** [mapeditor/tiled#2997] [Ideas]
I have this sprite sheet that I'm working on as I go. There can be dozens of characters on it with their animations.
I will start setting up their animation in tiled, and every now and then I need to modify the layout of my tile sheet.
Usually I just need to offset my tiles to make space for more.

It would be nice to have a way to shift (or drag move) animation data in the animation editor. As of now, every time I shift something on my tile sheet, I have to delete the affected animations 1 by 1 and recreate them all from scratch.

**Comment by bjorn** [comment]
> I will start setting up their animation in tiled, and every now and then I need to modify the layout of my tile sheet.
> Usually I just need to offset my tiles to make space for more.

This sounds related to issue #2863, since if you could offset the tiles while keeping their IDs the same, you wouldn't need to adjust your animations afterwards.

That said, the current animation editor is definitely a bit clumsy to work with. It's an area of improvement, but I'm not sure about the priority since it's a bit out of scope for a "level editor". If we do get to redesigning the animation editor I'd want to find a way to avoid the separate dialog for it as well. I think it could be done by displaying the selected tile's animation in a docked view, and having an explicit "animation editing" mode that overrides the tile selection (essentially like the terrain editing mode does as well).

**Comment by eishiya** [comment]
I realise this is rather late, but I only saw this post today, and decided to have a go at implementing this functionality as a script.

https://github.com/eishiya/tiled-scripts/blob/main/CopyPasteAnimations.js

This script lets you copy+paste animation data between tiles. In its default "Adjust Position" mode, it'll shift the pasted frames over, which should be just what you need when you're moving tiles around in a tilesheet.
